src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -738,6 +741,9 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                                 throw WinHttpException.CreateExceptionUsingError(lastError);
                             }
                         }
+
```

Comment:
>If there's an error in any of this initialization, do we want to dispose of the session handle

I'm assume you're talking about the stack variable `sessionHandle`...

I could wrap everything in a `finally` block and Dispose() it.  But it's not really worth it.  The session handle doesn't hold any network objects and it's just a small memory block that doesn't need to get cleaned up right away.  And it would only happen in obscure error cases when a session handle can't be created in the first place.